### PR TITLE
add systemlog api

### PR DESCRIPTION
**Pull request by yutongp**

### CODE REVIEW

File: proto/types/action.proto [hunk 1]
```diff
@@ -277,28 +277,47 @@ message Logs {
...
 
+// Deprecated
 message BlockEvmTransfer {
   uint64 blockHeight = 1;
   uint64 numEvmTransfers = 2;
   repeated ActionEvmTransfer actionEvmTransfers = 3;
 }
 
+message ActionSystemLog {
+  message Operation {
```
Comment (raullenchai) [inline proto/types/action.proto]: so no need for "from" and "to"?

is "address" == "to"?

File: proto/types/action.proto [hunk 2]
```diff
@@ -277,28 +277,47 @@ message Logs {
...
+    string amount = 1;  // amount can be negative to reflect out going balance
+    string address = 2;
+  }
+  bytes actionHash = 1;
+  uint64 numOperations = 2;
+  repeated Operation operations = 3;
+}
+
+message BlockSystemLog {
+  uint64 numOperations = 1;
```
Comment (raullenchai) [inline proto/types/action.proto]: blockHeight?

Comment (yutongp) [reply proto/types/action.proto]: see GetSystemLogByBlockHeightResponse

File: proto/types/action.proto (hunk 1 above)
Comment (yutongp) [reply proto/types/action.proto]: this is refer from rosetta API, they break down transaction into multiple operation. each operation contains one address and one positive/negative amount.
good part is this is very flexible, you can have 1to1 1tomany, single sided transaction.
down side is this introduce negative amount.

**Review by raullenchai — APPROVED** [review]

File: proto/types/action.proto [hunk 3]
```diff
@@ -277,28 +277,48 @@ message Logs {
...
 }
 
+message ActionSystemLog {
+  message Transaction {
+    string amount = 1;
+    string sender = 2;
+    string recipient = 3;
+  }
+  bytes actionHash = 1;
+  uint64 numOperations = 2;
```
Comment (dustinxie) [inline proto/types/action.proto]: nit: numTransactions?

**Pull request merged**